doc/sphinx/language/core/basic.rst

```diff
@@ -523,24 +523,24 @@ they appear after a boldface label.  They are listed in the
 Locality attributes supported by :cmd:`Set` and :cmd:`Unset`
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The :cmd:`Set` and :cmd:`Unset` commands support the :attr:`local`,
-:attr:`global` and :attr:`export` locality attributes:
-
-* no attribute: the original setting is *not* restored at the end of
-  the current module or section.
-* :attr:`local` (or alternatively, the ``Local`` prefix): the setting
+The :cmd:`Set` and :cmd:`Unset` commands support the mutually
+exclusive :attr:`local`, :attr:`export` and :attr:`global` locality
+attributes:
+
+* if no attribute is specified: the original value of the setting is
+  restored at the end of the current module but it is *not* restored
+  at the end of the current section.
+* :attr:`local` (or the ``Local`` prefix): the setting
   is applied within the current module or section.  The original value
   of the setting is restored at the end of the current module or
   section.
-* :attr:`export` (or alternatively, the ``Export`` prefix): similar to
-  :attr:`local`, the original value of the setting is restored at the
-  end of the current module or section.  In addition, if the value is
-  set in a module, then :cmd:`Import`\-ing the module sets the option
-  or flag.
-* :attr:`global` (or alternatively, the ``Global`` prefix): the
-  original setting is *not* restored at the end of the current module
-  or section.  In addition, if the value is set in a file, then
-  :cmd:`Require`\-ing the file sets the option.
+* :attr:`export` (or the ``Export`` prefix): same as no attribute, but
+  in addition, exported settings in :cmd:`Import`\ed modules override
+  the current value.
+* :attr:`global` (or the ``Global`` prefix): the original value of the
+  setting is *not* restored at the end of the current module or
+  section.  In addition, global settings in a file override the
+  current value when the file is :cmd:`Require`\d (even transitively).
```

Comment:
I notice you refer to files here and to modules above (line 538).  Is there a difference?  Maybe worth mentioning/repeating the distinction here (if there is one)?

Reply:
Yes, you can `Import` a module but only `Require` a file. Because a file is also a module, you can `Require Import` it. This whole chapter needs to be improved and consolidated anyway to make this fact clearer.